src/solver/impls/imex-bdf2/imex-bdf2.cxx

```diff
@@ -166,37 +148,36 @@ int IMEXBDF2::init(int nout, BoutReal tstep) {
   }
 
   // Allocate memory and initialise structures
-  u = new BoutReal[nlocal];
+  u = Array<BoutReal>{nlocal};
   for(int i=0;i<maxOrder;i++){
-    uV.push_back(new BoutReal[nlocal]);
-    fV.push_back(new BoutReal[nlocal]);
-    //gV.push_back(new BoutReal[nlocal]);
+    uV.emplace_back(Array<BoutReal>{nlocal});
+    fV.emplace_back(Array<BoutReal>{nlocal});
     timesteps.push_back(timestep);
     uFac.push_back(0.0);
     fFac.push_back(0.0);
     gFac.push_back(0.0);
   }
 
-  rhs = new BoutReal[nlocal];
+  rhs = Array<BoutReal>{nlocal};
 
   OPTION(options, adaptive, true); //Do we try to estimate the error?
   OPTION(options, nadapt, 4); //How often do we check the error
   OPTION(options, dtMinFatal, 1.0e-10);
   OPTION(options, dtMax, out_timestep);
   OPTION(options, dtMin, dtMinFatal);
   if(adaptive){
-    err = new BoutReal[nlocal];
+    err = Array<BoutReal>{nlocal};
     OPTION(options, adaptRtol, 1.0e-3); //Target relative error
     OPTION(options, mxstepAdapt, mxstep); //Maximum no. consecutive times we try to reduce timestep
     OPTION(options, scaleCushUp, 1.5);
     OPTION(options, scaleCushDown, 1.0);
   }
 
   // Put starting values into u
-  saveVars(u);
+  saveVars(std::begin(u));
```

Comment:
Presumably at some point saveVars et al could be modified to take an Array as input?